<documentation title="Lowercase Built-In functions">
    <standard>
    <![CDATA[
    All PHP built-in functions should be lowercased when called.
    ]]>
    </standard>
    <code_comparison>
        <code title="Valid: Lowercase function call.">
        <![CDATA[
if (<em>isset</em>($foo)) {
    echo $foo;
}
]]>
        </code>
        <code title="Invalid: isset not called as lowercase.">
        <![CDATA[
if (<em>isSet</em>($foo)) {
    echo $foo;
}
]]>
        </code>
    </code_comparison>
</documentation>
