<documentation title="Side Effects">
    <standard>
    <![CDATA[
    A php file should either contain declarations with no side effects, or should just have logic (including side effects) with no declarations.
    ]]>
    </standard>
    <code_comparison>
        <code title="Valid: A class defined in a file by itself.">
        <![CDATA[
<?php
class Foo
{
}
        ]]>
        </code>
        <code title="Invalid: A class defined in a file with other code.">
        <![CDATA[
<?php
class Foo
{
}

<em>echo "Class Foo loaded."</em>
        ]]>
        </code>
    </code_comparison>
</documentation>
