<documentation title="Closing Brace Indentation">
    <standard>
    <![CDATA[
    Closing braces should be indented at the same level as the beginning of the scope.
    ]]>
    </standard>
    <code_comparison>
        <code title="Valid: Consistent indentation level for scope.">
        <![CDATA[
if ($test) {
    $var = 1;
}
        ]]>
        </code>
        <code title="Invalid: The ending brace is indented further than the if statement.">
        <![CDATA[
if ($test) {
    $var = 1;
<em>    </em>}
        ]]>
        </code>
    </code_comparison>
</documentation>
