<documentation title="Closure Linter">
    <standard>
    <![CDATA[
    All javascript files should pass basic Closure Linter tests.
    ]]>
    </standard>
    <code_comparison>
        <code title="Valid: Valid JS Syntax is used.">
        <![CDATA[
var foo = [1, 2<em></em>];
        ]]>
        </code>
        <code title="Invalid: Trailing comma in a javascript array.">
        <![CDATA[
var foo = [1, 2<em>,</em>];
        ]]>
        </code>
    </code_comparison>
</documentation>
