<documentation title="Opening Tag at Start of File">
    <standard>
    <![CDATA[
    The opening php tag should be the first item in the file.
    ]]>
    </standard>
    <code_comparison>
        <code title="Valid: A file starting with an opening php tag.">
        <![CDATA[
<em></em><?php
echo 'Foo';
        ]]>
        </code>
        <code title="Invalid: A file with content before the opening php tag.">
        <![CDATA[
<em>Beginning content</em>
<?php
echo 'Foo';
        ]]>
        </code>
    </code_comparison>
</documentation>
