<documentation title="Closing PHP Tags">
    <standard>
    <![CDATA[
    All opening php tags should have a corresponding closing tag.
    ]]>
    </standard>
    <code_comparison>
        <code title="Valid: A closing tag paired with it's opening tag.">
        <![CDATA[
<em><?php</em>
echo 'Foo';
<em>?></em>
        ]]>
        </code>
        <code title="Invalid: No closing tag paired with the opening tag.">
        <![CDATA[
<em><?php</em>
echo 'Foo';
        ]]>
        </code>
    </code_comparison>
</documentation>
