<documentation title="Deprecated Functions">
    <standard>
    <![CDATA[
    Deprecated functions should not be used.
    ]]>
    </standard>
    <code_comparison>
        <code title="Valid: A non-deprecated function is used.">
        <![CDATA[
$foo = <em>explode</em>('a', $bar);
        ]]>
        </code>
        <code title="Invalid: A deprecated function is used.">
        <![CDATA[
$foo = <em>split</em>('a', $bar);
        ]]>
        </code>
    </code_comparison>
</documentation>
