<documentation title="Lowercase Keywords">
    <standard>
    <![CDATA[
    All PHP keywords should be lowercase.
    ]]>
    </standard>
    <code_comparison>
        <code title="Valid: Lowercase array keyword used.">
        <![CDATA[
$foo = <em>array</em>();
        ]]>
        </code>
        <code title="Invalid: Non-lowercase array keyword used.">
        <![CDATA[
$foo = <em>Array</em>();
        ]]>
        </code>
    </code_comparison>
</documentation>
