<documentation title="Echoed Strings">
    <standard>
    <![CDATA[
    Simple strings should not be enclosed in parentheses when being echoed.
    ]]>
    </standard>
    <code_comparison>
        <code title="Valid: Using echo without parentheses.">
        <![CDATA[
echo<em> </em>"Hello";
]]>
        </code>
        <code title="Invalid: Using echo with parentheses.">
        <![CDATA[
echo<em>(</em>"Hello"<em>)</em>;
]]>
        </code>
    </code_comparison>
</documentation>
