<documentation title="Space After Casts">
    <standard>
    <![CDATA[
    Exactly one space is allowed after a cast.
    ]]>
    </standard>
    <code_comparison>
        <code title="Valid: A cast operator is followed by one space.">
        <![CDATA[
$foo = (string)<em> </em>1;
        ]]>
        </code>
        <code title="Invalid: A cast operator is not followed by whitespace.">
        <![CDATA[
$foo = (string)<em></em>1;
        ]]>
        </code>
    </code_comparison>
</documentation>
